axelrod/strategies/ann.py

```diff
@@ -0,0 +1,222 @@
+        # These are True/False 0/1
+        if len(opponent.history) == 0:
+            opponent_first_c = 0
+            opponent_first_d = 0
+            opponent_second_c = 0
+            opponent_second_d = 0
+            my_previous_c = 0
+            my_previous_d = 0
+            my_previous2_c = 0
+            my_previous2_d = 0
+            opponent_previous_c = 0
+            opponent_previous_d = 0
+            opponent_previous2_c = 0
+            opponent_previous2_d = 0
+
+        elif len(opponent.history) == 1:
+            opponent_first_c = 1 if opponent.history[0] == C else 0
+            opponent_first_d = 1 if opponent.history[0] == D else 0
+            opponent_second_c = 0
+            opponent_second_d = 0
+            my_previous_c = 1 if self.history[-1] == C else 0
+            my_previous_d = 0 if self.history[-1] == D else 0
+            my_previous2_c = 0
+            my_previous2_d = 0
+            opponent_previous_c = 1 if opponent.history[-1] == C else 0
+            opponent_previous_d = 1 if opponent.history[-1] == D else 0
+            opponent_previous2_c = 0
+            opponent_previous2_d = 0
+
+        else:
+            opponent_first_c = 1 if opponent.history[0] == C else 0
+            opponent_first_d = 1 if opponent.history[0] == D else 0
+            opponent_second_c = 1 if opponent.history[1] == C else 0
+            opponent_second_d = 1 if opponent.history[1] == D else 0
+            my_previous_c = 1 if self.history[-1] == C else 0
+            my_previous_d = 0 if self.history[-1] == D else 0
+            my_previous2_c = 1 if self.history[-2] == C else 0
+            my_previous2_d = 1 if self.history[-2] == D else 0
+            opponent_previous_c = 1 if opponent.history[-1] == C else 0
+            opponent_previous_d = 1 if opponent.history[-1] == D else 0
+            opponent_previous2_c = 1 if opponent.history[-2] == C else 0
+            opponent_previous2_d = 1 if opponent.history[-2] == D else 0
+
+        # Remaining Features
+        turns_remaining = self.match_attributes['length'] - len(self.history)
+        total_opponent_c = opponent.history.count(C)
+        total_opponent_d = opponent.history.count(D)
+        total_self_c = self.history.count(C)
+        total_self_d = self.history.count(D)
+
+        output = self.activate([
+            opponent_first_c,
+            opponent_first_d,
+            opponent_second_c,
+            opponent_second_d,
+            my_previous_c,
+            my_previous_d,
+            my_previous2_c,
+            my_previous2_d,
+            opponent_previous_c,
+            opponent_previous_d,
+            opponent_previous2_c,
+            opponent_previous2_d,
+            total_opponent_c,
+            total_opponent_d,
+            total_self_c,
+            total_self_d,
+            turns_remaining
+        ])
+        if output > 0:
+            return C
+        else:
+            return D
+
+
+class EvolvedANN(ANN):
```

Comment:
Can we have more of a docstring here please. 

- Saying that it's based on a neural net etc.
- Include:

    ```
    Names:

         - Evolved ANN: : Original name by Marc Harper/Marting Jones
    ```